### PR TITLE
gomod parsers tests

### LINKED ISSUE #52: Tests for gomod parsers
After #2 is done
Tests for parsers need to be implemented

### PR DESCRIPTION
**Pull request by makhov**
Some tests added.
Fixes #52

### CODE REVIEW

**Comment by adelowo** [comment]
Oh... I previously PR'd #71 earlier anyways

**Comment by bketelsen** [comment]
oops - do you want to take a look and see if there's value in combining the two test PR's into one?

**Comment by adelowo** [comment]
I am currently on mobile and can't even view the changes conveniently...  Would take a look as soon as I get home - if this isn't merged by then.

**Comment by makhov** [comment]
Wow.
Let's combine the two PR's.
@adelowo found a lot of good negative cases, noticed unnecessary regex compilation and so on, but I think my tests a little bit simpler to read and understand.

I'll commit some changes soon (and think about zip files).

**Comment by makhov** [comment]
Hope I made it better (but not sure)

**Comment by adelowo** [comment]
Looks good enough.... I think this is much more better than #71 .
It is just left for @arschles and @michalpristas to make a choice

**Comment by arschles** [comment]
I'll need to take a look at this and #71 tomorrow before I can throw in my 2 cents 😄

File: pkg/gomod/bytes/parser_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	a := assert.New(t)
+
+	var testCases = []struct {
+		content     []byte
+		expected    string
+		expectedErr error
+	}{
+		{[]byte(`module "my/thing"`), "my/thing", nil},
+		{[]byte(`module "my/thing2`), "", parser.ErrNotFound},
+		{[]byte(`foobar`), "", parser.ErrNotFound},
```
Comment (michalpristas) [inline pkg/gomod/bytes/parser_test.go]: try moultiple modules in one file, first should be selected, don't know how vgo behaves in this case

File: pkg/gomod/file/parser_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+func TestFileParser_ModuleName(t *testing.T) {
+	a := assert.New(t)
+
+	var testCases = []struct {
+		file        string
+		expected    string
+		expectedErr error
+	}{
+		{"testdata/go.0.mod", "my/thing", nil},
```
Comment (michalpristas) [inline pkg/gomod/file/parser_test.go]: 💯 for testdata approach

File: pkg/gomod/file/testdata/go.0.mod [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module "my/thing"
```
Comment (michalpristas) [inline pkg/gomod/file/testdata/go.0.mod]: maybe one more with different order, parser should look it up

File: pkg/gomod/parser.go [hunk 4]
```diff
@@ -1,47 +1,51 @@
...
+)
+
+const (
+	moduleRegexp = `module "([\w\.@:%_\+-.~#?&]+/.+)"`
+)
+
+var (
+	re = regexp.MustCompile(moduleRegexp)
+	// ErrNotFound describe an error when can't find module name
+	ErrNotFound = errors.New("name not found")
```
Comment (michalpristas) [inline pkg/gomod/parser.go]: 👍

File: pkg/gomod/file/testdata/go.0.mod (hunk 3 above)
Comment (makhov) [reply pkg/gomod/file/testdata/go.0.mod]: Fixed in https://github.com/gomods/athens/pull/72/files#diff-ae9bcd418668c90e559d1c8331670d4c

File: pkg/gomod/bytes/parser_test.go (hunk 1 above)
Comment (makhov) [reply pkg/gomod/bytes/parser_test.go]: Thanks! Added in https://github.com/gomods/athens/pull/72/files#diff-6d0b6ed72e589e371bc00e27c79d5d9cR28

**Comment by michalpristas** [comment]
after last commits, perfect

**Comment by makhov** [comment]
And thanks to @adelowo for his work!

**Pull request merged**